torch/utils/data/_utils/worker.py

```diff
@@ -220,9 +221,22 @@ def _worker_loop(dataset_kind, dataset, index_queue, data_queue, done_event,
         signal_handling._set_worker_signal_handlers()
 
         torch.set_num_threads(1)
+
+        # Note [RNG re-seeding in Dataloader workers]
+        # We need to explicitly re-seed all RNGs here (builtin random, numpy, torch) to ensure that the RNGs are
+        # different across workers.
+        # We have to seed the default RNG and the user-made ones differently otherwise the RNG of all Generators
+        # would be the same as the default one.
```

Comment:
Actually, it's not obvious to me why you need to special case the default generator. If you always do `_generate_seed(generator) + seed`, shouldn't that keep the default generator having a different RNG than the others?

Reply:
The problem is that doing this for the default generator would break the contract that its seed should only depend on the `generator` that was passed as parameter to `Dataloader(..., generator=generator)` (by default, this is the default generator itself, but it could be another one).

That's why I didn't want to change the re-seeding logic of the default generator